hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/RDBBatchOperation.java

```diff
@@ -18,18 +18,225 @@
  */
 package org.apache.hadoop.hdds.utils.db;
 
+import com.google.common.base.Preconditions;
 import org.apache.hadoop.hdds.utils.db.RocksDatabase.ColumnFamily;
 import org.apache.hadoop.hdds.utils.db.managed.ManagedWriteBatch;
 import org.apache.hadoop.hdds.utils.db.managed.ManagedWriteOptions;
+import org.apache.ratis.util.StringUtils;
+import org.apache.ratis.util.TraditionalBinaryPrefix;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Supplier;
 
 /**
  * Batch operation implementation for rocks db.
  */
 public class RDBBatchOperation implements BatchOperation {
```

Comment:
It can be a good idea to put a comment stating this is not thread-safe and designed for single-thread use.

Reply:
Note that this is an existing class (not added by this change).  However, we could add the javadoc here.

Reply:
Thanks for updating it. Before this class is just a proxy to `org.rocksdb.WriteBatch`, but now it has some internal logic that's not intended for multi-thread.

Reply:
Hey, `org.rocksdb.WriteBatch` itself is not thread safe, see https://github.com/facebook/rocksdb/wiki/Basic-Operations
> However other objects (like Iterator and WriteBatch) may require external synchronization. If two threads share such an object, they must protect access to it using their own locking protocol.

Indeed, making a *batch* operation thread safe does not sound right since a batch is supposed to be sequential.  Making it thread safe decreases the performance.